astropy/utils/console.py

```diff
@@ -899,73 +899,80 @@ def __init__(self, msg, color='default', file=None, step=1,
         self._msg = msg
         self._color = color
         self._file = file
-        self._step = step
+        self._step = step + 1
```

Comment:
Why do add `+ 1` here?  Since the default step is 1 this means by default it will only update every 2 steps.

Reply:
The loop on the iterator does self._cur_iter % self._step, so if step is 1 the method will update every 'other' cycle. It was my attempt at emulating the generator. Though, I don't think I need to add a 1 here, it can just be accounted for in the loop

Reply:
I see now.  If you ask me the original semantics of this argument are actually a bug too.  The docs say "only update the spinner every {step} steps".  So if `step=1` that should mean update the spinner every 1 step, i.e., every step.  The current semantics with `step=1` mean it updates every 2 steps.  Unless I'm missing something that doesn't make any sense.

Reply:
In which case it should be that `step >= 1` and `step=1` should be the default.

Reply:
In fact I see the default is already `step=1`.  I think this was just a minor bug in the original code.  It should have read:

```python
for _ in range(self._step - 1):
    yield
```

Reply:
Opened an issue at #11656